config/formats.js

```diff
@@ -497,17 +497,7 @@ exports.Formats = [
 			'Rayquaza', 'Regigigas', 'Reshiram', 'Shaymin-Sky', 'Shedinja', 'Slaking', 'Solgaleo', 'Terrakion', 'Xerneas', 'Yveltal', 'Zekrom',
 			'Arena Trap', 'Power Construct', 'Shadow Tag', 'Gengarite', 'Kangaskhanite', 'Lucarionite', 'Salamencite', 'Baton Pass',
 		],
-		onValidateSet: function (set) {
-			let bannedAbilities = {'Arena Trap': 1, 'Comatose': 1, 'Contrary': 1, 'Fluffy': 1, 'Fur Coat': 1, 'Huge Power': 1, 'Illusion': 1, 'Imposter': 1, 'Innards Out': 1, 'Parental Bond': 1, 'Protean': 1, 'Pure Power': 1, 'Shadow Tag':1, 'Simple':1, 'Speed Boost': 1, 'Stakeout': 1, 'Water Bubble': 1, 'Wonder Guard': 1};
-			if (set.ability in bannedAbilities) {
```

Comment:
You don’t need to move this to the validator. Just remove the redundant declaration and replace `bannedAbilities` with `this.bannedAbilities`.

Reply:
https://i.imgur.com/XXnP7Pu.png

Reply:
Ah, I was remembering that wrong.

Reply:
Ok, if you add `, format` to the parameter list and change the old line 501 to `let bannedAbilities = format.bannedAbilities || []`, this should let you keep this validation in formats.js.